### PR TITLE
Added allowActionsInDocumentPreview property to MXKRoomViewController

### PR DESCRIPTION
**Pull request by gileluard**
This property is used for showing or hiding the actions button in document preview. Part of fix for https://github.com/vector-im/element-ios/issues/3864 property

### CODE REVIEW

File: MatrixKit/Controllers/MXKPreviewViewController.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+ Copyright 2020 Vector Creations Ltd
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKPreviewViewController.h:2]:
```suggestion
 Copyright 2020 The Matrix.org Foundation C.I.C
```

We have 2 kinds of copyright. Everything under https://github.com/matrix-org is transferred to the foundation.

File: MatrixKit/Controllers/MXKPreviewViewController.m [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+    
+    return self;
+}
+
+- (void)viewDidLayoutSubviews
+{
+    [super viewDidLayoutSubviews];
+    
+    if (!self.allowActions) {
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKPreviewViewController.m:72]:
```suggestion
    if (!self.allowActions)
    {
```

File: MatrixKit/Controllers/MXKPreviewViewController.m [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        NSMutableArray *items = [NSMutableArray arrayWithArray: self.previewController.navigationItem.rightBarButtonItems];
+        if (items.count > 0)
+        {
+            [items removeObjectAtIndex:0];
+        }
+        self.previewController.navigationItem.rightBarButtonItems = items;
+    }
+}
+
+- (IBAction)doneAction:(id)sender {
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKPreviewViewController.m:82]:
```suggestion
- (IBAction)doneAction:(id)sender
{
```

File: MatrixKit/Controllers/MXKRoomViewController.m [hunk 4]
```diff
@@ -3708,19 +3715,30 @@ - (void)showAttachmentInCell:(UITableViewCell*)cell
...
-                        self->documentInteractionController = [UIDocumentInteractionController interactionControllerWithURL:fileURL];
-                        [self->documentInteractionController setDelegate:self];
-                        self->currentSharedAttachment = selectedAttachment;
-
-                        if (![self->documentInteractionController presentPreviewAnimated:YES])
+                        if (self.allowActionsInDocumentPreview)
                         {
-                            if (![self->documentInteractionController presentOptionsMenuFromRect:self.view.frame inView:self.view animated:YES])
+                            // We could get rid of this part of code and use only a MXKPreviewViewController
+                            // Nevertheless, MXKRoomViewController is complient to UIDocumentInteractionControllerDelegate
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKRoomViewController.m:3721]:
```suggestion
                            // Nevertheless, MXKRoomViewController is compliant to UIDocumentInteractionControllerDelegate
```

**Comment by gileluard** [comment]
Simulator Screen Shot - iPad (8th generation) - 2020-12-09 at 10 03 54
Simulator Screen Shot - iPad (8th generation) - 2020-12-09 at 10 04 12
Simulator Screen Shot - iPhone 11 - 2020-12-09 at 10 07 29
Simulator Screen Shot - iPhone 11 - 2020-12-09 at 10 07 48

**Pull request merged**